### PR TITLE
OTG

### PR DESCRIPTION
**Pull request by burrbull**

### CODE REVIEW

**Comment by burrbull** [comment]
cc @Disasm

**Comment by burrbull** [comment]
@newAM Could you look at this?

File: devices/common_patches/usb_otg/fs_v1ext.yaml [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+      addressOffset: 0x250
+    HCCHAR11:
+      _from: HCCHAR0
+      addressOffset: 0x260
+    HCINT11:
+      _from: HCINT0
+      addressOffset: 0x268
+    HCINTMSK11:
+      _from: HCINTMSK0
+      addressOffset: 0x26C
```
Comment (newAM) [inline devices/common_patches/usb_otg/fs_v1ext.yaml:56]: Where do the offsets in this section come from?

The offset I got from the STM32F4 RM for `OTG_HS_HCINTMSKx` is `Address offset: 0x50C + 0x20 * x`, and I can't figure out how to get 0x26C from that.

File: devices/common_patches/usb_otg/hs_fixes_446_469.yaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+      - SDET
+      - PS2DET
+      - BCDEN
+      - DCDEN
+      - PDEN
+      - SDEN
+
+  _modify:
+    CID:
+      resetValue: 0x2100
```
Comment (newAM) [inline devices/common_patches/usb_otg/hs_fixes_446_469.yaml:15]: image

Isn't this `0x1200`?

**Review by newAM — COMMENTED** [review]
Took a while to find time to review thing since it's so big :)

Looks good so far, just have a question about that one reset value and I'm not sure how some of the addressOffsets are calculated.

File: devices/common_patches/usb_otg/hs_fixes_446_469.yaml (hunk 2 above)
Comment (burrbull) [reply devices/common_patches/usb_otg/hs_fixes_446_469.yaml:15]: I see this in RM0386 Rev 5 and RM0390 Rev 6:
Screenshot_20230129_121419

File: devices/common_patches/usb_otg/fs_v1ext.yaml (hunk 1 above)
Comment (burrbull) [reply devices/common_patches/usb_otg/fs_v1ext.yaml:56]: You should take into account that OTG_FS_HOST is independent peripheral with offset of 0x400

So is actually `0x400 + 0x10C + 0x20 * x`

File: devices/common_patches/usb_otg/hs_fixes_446_469.yaml (hunk 2 above)
Comment (newAM) [reply devices/common_patches/usb_otg/hs_fixes_446_469.yaml:15]: My bad, I was looking at the F4x5 RM (RM0090 Rev 19), not the F446/F469 RM.

**Review by newAM — APPROVED** [review]
Sorry it took a while for me to get back to this one :)

Looks good to me!

bors r+

**Pull request merged**